lib/core/scheduler.hpp

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+#include "logger.hpp"
+#include "time.hpp"
+#include "types.hpp"
+
+#include <cstdint>
+#include <functional>
+#include <queue>
+#include <unordered_map>
+
+namespace hyped::core {
+
+struct Task {
```

Comment:
food for thought: maybe we want to have this live inside the `Scheduler` class, i.e. `core::Scheduler::Task`. I don't feel very strongly about this, but it's worth mentioning because as far as I can tell this struct will only be used inside the `Scheduler`.